UpdateService/UpdateService.cs

```diff
@@ -18,49 +17,43 @@ namespace Hypertherm.Update
     public interface IUpdateService
     {
         Task Update(string version = "latest");
-        Task<bool> IsUpdateAvailable();
+        Version LatestReleasedVersion();
+        bool IsUpdateAvailable();
         Task<List<string>> ListReleases();
     }
 
     public class UpdateWithGitHubAPI : IUpdateService
     {
+        private bool _updateIsAvailable = false;
         private string gitHubUrl;
         private IAnalyticsService _analyticsService;
         private ILoggingService _logger;
         private static HttpClient _httpClient;
 
         private List<JObject> _releases = new List<JObject>();
-        private string _latestReleaseUrl;
+
+        private Version _latestReleasedVersion = new Version("0.0.0.0");
+        public Version LatestReleasedVersion(){ return _latestReleasedVersion; }
+        private string _latestReleaseUrl = "";
 
         public UpdateWithGitHubAPI(IAnalyticsService analyticsService, ILoggingService logger, string url = "")
         {
             _analyticsService = analyticsService;
             _logger = logger;
             gitHubUrl = url;
             _httpClient = new HttpClient();
+
+            GetLatestReleasedVersionInfo().GetAwaiter().GetResult();
         }
 
-        public async Task<bool> IsUpdateAvailable()
+        public bool IsUpdateAvailable()
         {
-            _analyticsService.GenericTrace($"Checking for updates.");
-
-            var latestVersion = new Version("0.0.0.0");
+            _analyticsService.GenericTrace($"Checking for available updates.");

```

Comment:
```suggestion
            _analyticsService.GenericTrace($"Checking for updates.");

```

Reply:
Yea this looks good. I might even go as far as changing updates to singular update.
Keep in mind _analyticsService is a logging interface, for us it is using Application Insights.